externalip/pkg/operator/operator.go

```diff
@@ -261,7 +261,7 @@ func (op *Operator) syncHandler(_ interface{}) error {
 	// Make sure we keep stats for the top-level chains, if they existed
 	// (which most should have because we created them above).
 	if chain, ok := existingFilterChains[kubeExternalIPChain]; ok {
-		writeLine(filterChains, chain)
+		writeLine(filterChains, utiliptables.MakeChainLine(utiliptables.Chain(chain)))
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/iptables/iptables.go#L117
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/iptables/save_restore.go#L31

The code is equivalent to:
`fmt.Sprintf(":%s - [0:0]", String(chain))`

If all that happened is that `string` got turned into `byte[]` this will create:
```
::KUBE-SVC-XENJGWWDHWIOQZ4E - [0:0] - [0:0] 
```

I'd go with:
```
writeLine(filterChains, utiliptables.Chain(chain))
```

Reply:
Unfortunately this is not accepted due to type mismatch `pkg/operator/operator.go:264:45: cannot use iptables.Chain(chain) (type iptables.Chain) as type string in argument to writeLine`

I see that proxier.go was changed and would propose to go with the latest example
https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L912-L920

Though `WriteLine` and `WriteLineBytes` functions are not available in 1.15 in k8s.io/kubernetes/pkg/proxy/util (came with 1.21),  I will adjust them from:
https://github.com/kubernetes/kubernetes/blob/617064d7327bf455955cb785d57330c425d48b51/pkg/proxy/util/utils.go#L475-L491
https://github.com/kubernetes/kubernetes/blob/617064d7327bf455955cb785d57330c425d48b51/pkg/proxy/util/utils.go#L475-L491

Seems that we also win some performance by avoiding strings.Join